src/RecyclableMemoryStreamManager.cs

```diff
@@ -272,16 +300,20 @@ internal byte[] GetBlock()
 
         /// <summary>
         /// Returns a buffer of arbitrary size from the large buffer pool. This buffer
-        /// will be at least the requiredSize and always be a multiple of largeBufferMultiple.
+        /// will be at least the requiredSize and always be a multiple/exponential of largeBufferMultiple.
         /// </summary>
         /// <param name="requiredSize">The minimum length of the buffer</param>
         /// <param name="tag">The tag of the stream returning this buffer, for logging if necessary.</param>
         /// <returns>A buffer of at least the required size.</returns>
         internal byte[] GetLargeBuffer(int requiredSize, string tag)
         {
-            requiredSize = this.RoundToLargeBufferMultiple(requiredSize);
+            requiredSize = !this.useExponentialLargeBuffer
```

Comment:
Maybe hide this logic inside universal RoundToLargeBufferSize and GetPoolIndex methods.